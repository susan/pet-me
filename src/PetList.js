import React from "react";
import { Pet } from "./Pet";

export const PetList = ({ Animals }) => {
  const animalList =
    Animals &&
    Animals.map(pet => {
      return <Pet key={pet.id} pet={pet} />;
    });

  return <>{animalList}</>;
};
