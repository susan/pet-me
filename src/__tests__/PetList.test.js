import React from "react";
import { shallow } from "enzyme";
import { PetList } from "../PetList";

it("should render a PetList", () => {
  const mockPets = [
    {
      id: 45262286,
      organization_id: "CA216",
      url:
        "https://www.petfinder.com/dog/marta-45262286/ca/riverside/mary-s-roberts-pet-adoption-center-ca216/?referrer_id=3d19601b-bfbd-46a9-996a-60c4ce5f6448",
      type: "Dog",
      species: "Dog"
    },
    {
      id: 45262286,
      organization_id: "CA216",
      url:
        "https://www.petfinder.com/dog/marta-45262286/ca/riverside/mary-s-roberts-pet-adoption-center-ca216/?referrer_id=3d19601b-bfbd-46a9-996a-60c4ce5f6448",
      type: "Dog",
      species: "Dog"
    }
  ];

  const wrapper = shallow(<PetList pet={mockPets} />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.props()).toEqual({});
});
