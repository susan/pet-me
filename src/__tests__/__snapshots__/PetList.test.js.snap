// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a PetList 1`] = `<Fragment />`;
